src/status_im/contexts/wallet/swap/events.cljs

```diff
@@ -577,3 +587,82 @@
              [:dispatch
               [:navigate-to-within-stack
                [:screen/wallet.swap-select-asset-to-pay :screen/wallet.swap-select-account]]]])})))
+
+(rf/reg-event-fx :wallet/get-swap-proposal-fee
+ (fn [{:keys [db]} [{:keys [amount-in amount-out]}]]
+   (let [wallet-address          (get-in db [:wallet :current-viewing-account-address])
+         {:keys [asset-to-pay asset-to-receive
+                 network]}       (get-in db [:wallet :ui :swap])
+         test-networks-enabled?  (get-in db [:profile/profile :test-networks-enabled?])
+         networks                ((if test-networks-enabled? :test :prod)
+                                  (get-in db [:wallet :networks]))
+         network-chain-ids       (map :chain-id networks)
+         pay-token-decimal       (:decimals asset-to-pay)
+         pay-token-id            (:symbol asset-to-pay)
+         receive-token-id        (:symbol asset-to-receive)
+         receive-token-decimals  (:decimals asset-to-receive)
+         gas-rates               constants/gas-rate-medium
+         amount-in-hex           (if amount-in
+                                   (send-utils/amount-in-hex amount-in pay-token-decimal)
+                                   0)
+         amount-out-hex          (when amount-out
+                                   (send-utils/amount-in-hex amount-out receive-token-decimals))
+         to-address              wallet-address
+         from-address            wallet-address
+         swap-chain-id           (:chain-id network)
+         disabled-to-chain-ids   (filter #(not= % swap-chain-id) network-chain-ids)
+         disabled-from-chain-ids (filter #(not= % swap-chain-id) network-chain-ids)
+         from-locked-amount      {}
+         send-type               constants/send-type-swap
+         request-uuid            (str (random-uuid))
+         params                  [(cond->
```

Comment:
This is quite a duplication of functionality with `:wallet/start-get-swap-proposal` and it also duplicates all the "not-so-good-patterns". For example, `to-address` and `from-address` are useless variables that can be substituted with `wallet-address`. `disabled-to-chain-ids` and `disabled-from-chain-ids` are identical, and so on.

Instead of duplicating it we can create a separate function that will prepare parameters for `get-suggested-routes`. Something like `params-for-swap-routes-request` That function can get `db` as an argument and encapsulate all the commonalities between `:wallet/get-swap-proposal-fee` and `:wallet/start-get-swap-proposal`. All the differences can go in additional args.  Also as a bonus it will hide the list of `status-go` parameters from event. 
Wdyt?

Reply:
+1 on this. There's too much duplication with the router events.

Reply:
Sure let me see if i can transfer some logic to utils or in specific sub,

Reply:
I believe a function in the same file that has common logic between `:wallet/start-get-swap-proposal` and yours `:wallet/get-swap-proposal-fee` would be sufficient

Reply:
just added `get-swap-proposal-params` func to eliminate duplication. 
https://github.com/status-im/status-mobile/pull/22048/commits/e29bd1641cb4c8ed513caf5c7e01b908aa697994